app/services/activitypub/synchronize_followers_service.rb

```diff
@@ -50,21 +64,35 @@ def build_undo_follow_json(follow)
     Oj.dump(serialize_payload(follow, ActivityPub::UndoFollowSerializer))
   end
 
-  def collection_items(collection_or_uri)
+  # Yields to the block on each page
+  # Only returns true if the whole collection has been processed
+  def process_collection(collection_or_uri, max_pages: MAX_COLLECTION_PAGES)
```

Comment:
I do not see a case where this is called with something other than a URL/URI, so maybe the first parameter could just be called `uri` or `collection_uri`?